### ISSUE TITLE
`traceback.format_exception_only` incorrect on Python >= 3.10

### ISSUE THREAD

**Issue by Gollum999** [python/typeshed#10841]
Starting in Python 3.10, the `traceback` package changed the behavior of `format_exception_only` to accept an exception object in the first argument:

> Since Python 3.10, instead of passing _value_, an exception object can be passed as the first argument. If _value_ is provided, the first argument is ignored in order to provide backwards compatibility.

It looks like the typeshed was updated for `print_exception`/`format_exception`, but not `format_exception_only`.

Mypy example:

```python
import traceback
try:
    raise RuntimeError("foo")
except Exception as e:
    traceback.print_exception(e)
    print(traceback.format_exception(e))
    print(traceback.format_exception_only(e))
```
```
main.py:7: error: Argument 1 to "format_exception_only" has incompatible type "Exception"; expected "type[BaseException] | None"  [arg-type]
Found 1 error in 1 file (checked 1 source file)
```
[Playground link](https://mypy-play.net/?mypy=latest&python=3.10&flags=strict&gist=0fe6ed84590277ce14a57cb83249bc90)

**Comment by srittau** [comment]
This was actually fixed in typeshed a while back, but mypy 1.5.1 doesn't include that yet. I assume that the next mypy version will include the fix.

**Comment by aminalaee** [comment]
Checking it with mpy master show it was fixed: [Playground link](https://mypy-play.net/?mypy=latest&python=3.10&flags=strict&gist=0fe6ed84590277ce14a57cb83249bc90)
